canvas_sdk/client/base.py

```diff
@@ -105,24 +109,33 @@ def call(action, url, request_context, params=None, data=None, max_retries=None,
         try:
             # build and send the request
             response = canvas_session.request(
-                action, url, params=params, data=data, headers=headers, cookies=cookies,
-                files=files, auth=auth, timeout=timeout, proxies=proxies, verify=verify,
-                cert=cert, allow_redirects=allow_redirects)
+                action, url, params=params, data=data, headers=headers,
+                cookies=cookies, files=files, auth=auth, timeout=timeout,
+                proxies=proxies, verify=verify, cert=cert,
+                allow_redirects=allow_redirects)
 
             # raise an http exception if one occured
             response.raise_for_status()
 
         except HTTPError as http_error:
+            log.info("Caught an API Error returned by Canvas: %s", str(http_error))
             # Need to check its an error code that can be retried
-            status_code = http_error.response.status_code
+            status_code = response.status_code
```

Comment:
The `raise_for_status` method is the only place where an `HTTPError` will be raised, so assuming we have a response is safe for this except block.  There are a number of other exceptions that the requests library may raise while making the request though, and I'd suggest we revisit those and decide if/how we want to map them to SDK exceptions as part of a future release.